File: backend/recommendation_engine.py
"""
ReadyDay — Recommendation Engine
Responsabilidad única: decisión principal (copy corto de acción diaria).

  "Hoy entrena fuerte" / "Hoy entrena moderado" / "Hoy descansa"

Para insight, prediction y topFactors → explanation_engine.py
"""

from typing import Literal

Zone = Literal["green", "yellow", "red"]
Lang = Literal["es", "en"]


def get_recommendation(zone: Zone, snapshot: dict, lang: Lang = "es") -> str:
    """
    Texto de decisión principal.
    Puerto de decision(z, d) del HTML MVP.
    """
    e = lang == "es"
    activity = float(snapshot.get("activityLoad") or 0)

    if zone == "green":
        return "Hoy entrena fuerte" if e else "Train hard today"
    if zone == "yellow":
        if activity > 60:
            return "Hoy entrena suave" if e else "Go light today"
        return "Hoy entrena moderado" if e else "Train moderate today"
    return "Hoy descansa" if e else "Rest today"

File: mobile/domain/recommendationEngine.ts
/**
 * ReadyDay — Recommendation Engine (TypeScript)
 * Puerto de backend/recommendation_engine.py
 * Responsabilidad única: decisión principal (copy corto de acción diaria).
 *
 * Para insight, prediction y topFactors → explanationEngine.ts
 */

import type { Zone } from '../../shared-types/readiness';
import type { CleanSnapshot } from '../garmin/normalizeSnapshot';

export type Lang = 'es' | 'en';

/**
 * Devuelve la decisión corta del día.
 * Puerto de get_recommendation() de Python.
 *
 * "Hoy entrena fuerte" / "Hoy entrena moderado" / "Hoy entrena suave" / "Hoy descansa"
 */
export function getRecommendation(zone: Zone, snap: CleanSnapshot, lang: Lang = 'es'): string {
  const e = lang === 'es';
  const activity = snap.activityLoad;

  if (zone === 'green') return e ? 'Hoy entrena fuerte' : 'Train hard today';
  if (zone === 'yellow') {
    if (activity > 60) return e ? 'Hoy entrena suave' : 'Go light today';
    return e ? 'Hoy entrena moderado' : 'Train moderate today';
  }
  return e ? 'Hoy descansa' : 'Rest today';
}
